sdk-lib/src/main/java/cash/z/ecc/android/sdk/model/FiatCurrency.kt

```diff
@@ -0,0 +1,34 @@
+package cash.z.ecc.android.sdk.model
+
+import android.icu.util.Currency
+
+data class FiatCurrency(val code: String) {
+    val symbol: String
+        get() = Currency.getInstance(code).symbol
+
+    init {
+        check(isAlpha3Code(code))
+    }
+
+    companion object Factory {
+        val USD = FiatCurrency("USD")
+
+        /**
+         * Checks whether [code] complies with ISO 4217 3-letter code.
+         */
+        @Suppress("ReturnCount", "MagicNumber")
+        fun isAlpha3Code(code: String): Boolean {
```

Comment:
It's great to have this covered. Just a question: do we intentionally let this public? Or should we make it private?

Reply:
yes it was intentional :)